### PR TITLE
py-rioxarray: add v0.17.0

### PR DESCRIPTION
**Pull request by Chrismarsh**
Update to include all current available releases of py-rioxarray

### CODE REVIEW

**Comment by Chrismarsh** [comment]
@spackbot fix style

**Comment by adamjstewart** [comment]
Can I convince you to only add the versions that you need so this is easier to review?

**Comment by Chrismarsh** [comment]
Sure, no problem. I've pushed that change

**Review by bernhardkaindl — DISMISSED** [review]
Built successfully on Ubuntu 22.04

**Review by adamjstewart — CHANGES_REQUESTED** [review]
All dependencies still need to be updated, see the `pyproject.toml`

**Comment by Chrismarsh** [comment]
Thanks, @adamjstewart. I am not sure how I missed doing that, so thanks for the catch. I've cleaned up the versioning, please let me know if you are happy with this

File: var/spack/repos/builtin/packages/py-rioxarray/package.py [hunk 1]
```diff
@@ -16,13 +16,25 @@ class PyRioxarray(PythonPackage):
...
     depends_on("py-setuptools", type="build")
-    depends_on("py-rasterio", type=("build", "run"))
+
+    # for optional interp, let numpy control scipy version
     depends_on("py-scipy", type=("build", "run"))
-    depends_on("py-xarray@0.17:", type=("build", "run"))
-    depends_on("py-pyproj@2.2:", type=("build", "run"))
+
+    with when("@0.17.0"):
+        depends_on("python@3.10:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rioxarray/package.py:30]: Missing packaging dependency

File: var/spack/repos/builtin/packages/py-rioxarray/package.py [hunk 2]
```diff
@@ -16,13 +16,25 @@ class PyRioxarray(PythonPackage):
...
     version(
         "0.4.1.post0", sha256="f043f846724a58518f87dd3fa84acbe39e15a1fac7e64244be3d5dacac7fe62b"
     )
 
-    depends_on("python@3.7:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-    depends_on("py-rasterio", type=("build", "run"))
+
+    # for optional interp, let numpy control scipy version
     depends_on("py-scipy", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rioxarray/package.py]: If this is optional we should either add a variant or remove it

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-rioxarray/package.py (hunk 2 above)
Comment (Chrismarsh) [reply var/spack/repos/builtin/packages/py-rioxarray/package.py]: Ok. I have done so, and set it to true to continue the existing behaviour of this package.

File: var/spack/repos/builtin/packages/py-rioxarray/package.py (hunk 1 above)
Comment (Chrismarsh) [reply var/spack/repos/builtin/packages/py-rioxarray/package.py:30]: thankyou

File: var/spack/repos/builtin/packages/py-rioxarray/package.py [hunk 3]
```diff
@@ -16,13 +16,31 @@ class PyRioxarray(PythonPackage):
...
     license("Apache-2.0")
 
+    version("0.17.0", sha256="46c29938827fff268d497f7ae277077066fcfbac4e53132ed3d4e2b96455be62")
     version(
         "0.4.1.post0", sha256="f043f846724a58518f87dd3fa84acbe39e15a1fac7e64244be3d5dacac7fe62b"
     )
 
-    depends_on("python@3.7:", type=("build", "run"))
+    # interpolation variant
+    # default True to match existing behaviour of this package
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rioxarray/package.py]: I would rather default to False to match the behavior of pip and conda.

File: var/spack/repos/builtin/packages/py-rioxarray/package.py [hunk 4]
```diff
@@ -16,13 +16,31 @@ class PyRioxarray(PythonPackage):
...
+    variant("interp", default=True, when="@0.17.0:", description="Enable interpolation routines")
+
     depends_on("py-setuptools", type="build")
-    depends_on("py-rasterio", type=("build", "run"))
-    depends_on("py-scipy", type=("build", "run"))
-    depends_on("py-xarray@0.17:", type=("build", "run"))
-    depends_on("py-pyproj@2.2:", type=("build", "run"))
+    depends_on("py-wheel", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rioxarray/package.py]: Wheel is automatically added by the base class, you don't need to add it yourself.

File: var/spack/repos/builtin/packages/py-rioxarray/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-rioxarray/package.py:30]: Still missing

Comment (Chrismarsh) [reply var/spack/repos/builtin/packages/py-rioxarray/package.py:30]: It should actually be resolved now 🫠

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**